alchemy/src/test/vitest.ts

```diff
@@ -0,0 +1,200 @@
+
+  beforeAll(fn: (scope: Scope) => Promise<void>): void;
+
+  afterAll(fn: (scope: Scope) => Promise<void>): void;
+
+  /**
+   * Current test scope
+   */
+  scope: Scope;
+};
+
+/**
+ * Creates a test helper function that provides scoped resource management
+ *
+ * @param meta Import meta object from the test file
+ * @param defaultOptions Default options to apply to all tests
+ * @returns Test function with scope management
+ *
+ * @example
+ * ```typescript
+ * const test = alchemy.test(import.meta, {
+  prefix: BRANCH_PREFIX
+});
+ *
+ * describe("My Resource", () => {
+ *   test("create and delete", async (scope) => {
+ *     try {
+ *       const resource = await MyResource("test", { ... });
+ *       expect(resource.id).toBeTruthy();
+ *     } finally {
+ *       await alchemy.destroy(scope);
+ *     }
+ *   });
+ * });
+ * ```
+ */
+export function test(meta: ImportMeta, defaultOptions?: TestOptions): test {
+  defaultOptions = defaultOptions ?? {};
+  if (
+    defaultOptions.stateStore === undefined &&
+    process.env.ALCHEMY_STATE_STORE === "cloudflare"
+  ) {
+    defaultOptions.stateStore = (scope) =>
+      new R2RestStateStore(scope, {
+        apiKey: alchemy.secret(process.env.CLOUDFLARE_API_KEY),
+        email: process.env.CLOUDFLARE_EMAIL,
+        bucketName: process.env.CLOUDFLARE_BUCKET_NAME!,
+      });
+  }
+
+  test.skipIf = it.skipIf.bind(it);
+
+  const scope = new Scope({
+    scopeName: `${defaultOptions.prefix ? `${defaultOptions.prefix}-` : ""}${path.basename(meta.filename)}`,
+    stateStore: defaultOptions?.stateStore,
+    phase: "up",
+  });
+
+  test.beforeAll = (fn: (scope: Scope) => Promise<void>) => {
+    return beforeAll(() => scope.run(() => fn(scope)));
+  };
+
+  test.afterAll = (fn: (scope: Scope) => Promise<void>) => {
+    return afterAll(() => scope.run(() => fn(scope)));
+  };
+
+  test.scope = scope;
+
+  return test as unknown as test;
+
+  function test(
+    ...args:
+      | [
+          name: string,
+          options: TestOptions,
+          fn: (scope: Scope) => Promise<void>,
+        ]
+      | [name: string, fn: (scope: Scope) => Promise<void>]
+  ) {
+    const testName = args[0];
+    const _options = typeof args[1] === "object" ? args[1] : undefined;
+    const timeout =
+      typeof args[args.length - 1] === "number"
+        ? (args[args.length - 1] as number)
+        : 120000;
+    const spread = (obj: any) =>
+      obj && typeof obj === "object"
+        ? Object.fromEntries(
+            Object.entries(obj).flatMap(([k, v]) =>
+              v !== undefined ? [[k, v]] : [],
+            ),
+          )
+        : {};
+
+    const options: TestOptions = {
+      quiet: false,
+      password: "test-password",
+      ...spread(defaultOptions),
+      ...spread(_options),
+    };
+
+    const fn = typeof args[1] === "function" ? args[1] : args[2]!;
+
+    return it(
+      testName,
+      (_ctx) => {
+        return alchemy.run(
+          testName,
```

Comment:
merge from main and restore this from git. It does not have al the same content as what is in main.